x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeToCharProcessorTests.java

```diff
@@ -0,0 +1,134 @@
+     *     <li>Spin up a Postgres instance (latest or a specific version) using docker:
+     *       <pre>
+     *       docker run --rm --name postgres-latest -e POSTGRES_PASSWORD=mysecretpassword -p 5432:5432 -d postgres:latest
+     *       </pre>
+     *     </li>
+     *     <li>Generate the test dataset by execution the SQL against PostgreSQL and capturing the output:
+     *       <pre>
+     *       PGPASSWORD="mysecretpassword" psql --quiet -h localhost -p 5432 -U postgres -f /tmp/postgresql-tochar-test.sql \
+     *           &gt; /path/to/tochar.generated.csv
+     *       </pre>
+     *     </li>
+     * </ol>
+     */
+    public void testAgainstPostgreSQLOutput() throws Exception {
+        String testFile = "tochar.generated.csv";
+        Path testFilePath = Path.of(DateTimeToCharProcessorTests.class.getResource(testFile).toURI());
+        int lineNumber = 0;
+        for (String line : Files.readAllLines(testFilePath)) {
+            lineNumber += 1;
+            if (line.startsWith("#")) {
+                continue;
+            }
+            String[] cols = line.split(quote(ToCharTestScript.DELIMITER));
+            testOneCase(testFile, lineNumber, cols[0], cols[1], cols[2], cols[3], cols[4]);
+        }
+    }
+
+    /**
+     * @param testFile The name of the testfile where this testcase is coming from
+     * @param lineNumber The line number of the testcase within the testfile
+     * @param secondsAndFractionsSinceEpoch The date represented by seconds and fractions since epoch that was used to 
+     *                                      generate the TO_CHAR() PostgreSQL output.
+     * @param zone The long/short name or offset for the timezone used when generating the expected TO_CHAR() output.
+     * @param formatString The pattern to be tested (this is exactly the pattern that was passed into the TO_CHAR() function in PostgreSQL).
+     * @param posgresTimestamp The timestamp represented by PostgreSQL as string in the default format (without calling TO_CHAR()).
+     * @param expectedResult The PostgreSQL output of <code>TO_CHAR(
+     *                       (TO_TIMESTAMP([[secondsSinceEpoch]]) + INTERVAL '[[fractions]] microseconds'), 
+     *                       '[[formatString]]')</code>.
+     */
+    private void testOneCase(
+        String testFile, int lineNumber, String secondsAndFractionsSinceEpoch, String zone,
+        String formatString, String posgresTimestamp, String expectedResult) {
+        ZoneId zoneId = ZoneId.of(zone);
+        ZonedDateTime timestamp = dateTimeWithFractions(secondsAndFractionsSinceEpoch);
+        String actualResult =
+            (String) new ToChar(Source.EMPTY, l(timestamp, DataTypes.DATETIME), l(formatString, DataTypes.KEYWORD), zoneId)
+                .makePipe()
+                .asProcessor()
+                .process(null);
+        List<String> expectedResultSplitted = asList(expectedResult.split(quote(ToCharTestScript.PATTERN_DELIMITER)));
```

Comment:
I think all these lines would be more readable if you'd `import static` all these static types.

Reply:
Done.